clients/admin-ui/src/pages/messaging/index.tsx

```diff
@@ -0,0 +1,275 @@
+import { CustomizableMessagingTemplatesEnum } from "~/features/messaging-templates/CustomizableMessagingTemplatesEnum";
+import CustomizableMessagingTemplatesLabelEnum from "~/features/messaging-templates/CustomizableMessagingTemplatesLabelEnum";
+import {
+  useGetSummaryMessagingTemplatesQuery,
+  usePatchMessagingTemplateByIdMutation,
+} from "~/features/messaging-templates/messaging-templates.slice";
+import { MessagingTemplateWithPropertiesSummary } from "~/types/api";
+import { isErrorResult } from "~/types/errors";
+
+const columnHelper =
+  createColumnHelper<MessagingTemplateWithPropertiesSummary>();
+
+const EMPTY_RESPONSE = {
+  items: [],
+  total: 0,
+  page: 1,
+  size: 50,
+  pages: 1,
+};
+
+const MessagingPage: NextPage = () => {
+  const router = useRouter();
+  const toast = useToast();
+
+  const {
+    PAGE_SIZES,
+    pageSize,
+    setPageSize,
+    onPreviousPageClick,
+    isPreviousPageDisabled,
+    onNextPageClick,
+    isNextPageDisabled,
+    startRange,
+    endRange,
+    pageIndex,
+    setTotalPages,
+  } = useServerSidePagination();
+
+  const {
+    data: templateResponse,
+    isLoading,
+    isFetching,
+  } = useGetSummaryMessagingTemplatesQuery({
+    page: pageIndex,
+    size: pageSize,
+  });
+
+  const [patchMessagingTemplateById] = usePatchMessagingTemplateByIdMutation();
+
+  const {
+    items: data,
+    total: totalRows,
+    pages: totalPages,
+  } = useMemo(() => templateResponse ?? EMPTY_RESPONSE, [templateResponse]);
+
+  useEffect(() => {
+    setTotalPages(totalPages);
+  }, [totalPages, setTotalPages]);
+
+  const [isAddTemplateModalOpen, setIsAddTemplateModalOpen] =
+    useState<boolean>(false);
+
+  const columns = useMemo(
+    () => [
+      columnHelper.accessor((row) => row.type, {
+        id: "message",
+        cell: (props) => (
+          <DefaultCell
+            value={
+              CustomizableMessagingTemplatesLabelEnum[
+                props.getValue() as CustomizableMessagingTemplatesEnum
+              ]
+            }
+          />
+        ),
+        header: (props) => <DefaultHeaderCell value="Message" {...props} />,
+        size: 150,
+      }),
+
+      columnHelper.accessor(
+        (row) => (row.properties || []).map((property) => property.name),
+        {
+          id: "properties",
+          cell: (props) => (
+            <GroupCountBadgeCell
+              suffix="properties"
+              value={props.getValue()}
+              {...props}
+            />
+          ),
+          header: (props) => (
+            <DefaultHeaderCell value="Properties" {...props} />
+          ),
+          meta: {
+            displayText: "Properties",
+            showHeaderMenu: true,
+          },
+          size: 250,
+        }
+      ),
+      columnHelper.accessor((row) => row.is_enabled, {
+        id: "is_enabled",
+        cell: (props) => (
+          <Flex align="center" justifyContent="flex-start" w="full" h="full">
```

Comment:
Would prefer for this to be extracted out to its own file if possible.

Reply:
Yes. I've extracted it into the useMessagingTemplateToggle hook. This reduces the amount of code & imports in the messaging index page nicely.